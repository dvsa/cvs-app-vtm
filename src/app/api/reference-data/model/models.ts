export * from './reference-data.model';
