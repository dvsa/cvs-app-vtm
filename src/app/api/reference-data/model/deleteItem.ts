/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}