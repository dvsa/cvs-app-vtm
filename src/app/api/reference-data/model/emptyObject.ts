/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
