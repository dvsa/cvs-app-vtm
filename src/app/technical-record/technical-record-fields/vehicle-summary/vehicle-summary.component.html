<div class="govuk-grid-row">
  <div class="govuk-grid-column-full">
    <div class="govuk-form-group">
      <fieldset class="govuk-fieldset">
        <div [formGroup]="technicalRecord">
          <h2 class="govuk-heading-m">Vehicle summary</h2>

          <div class="govuk-radios govuk-radios--inline">
            <label for="vehicleType" class="govuk-label govuk-!-font-weight-bold">
              Vehicle type
            </label>
            <div class="govuk-radios__item" *ngFor="let option of options">
              <input
                class="govuk-radios__input"
                id="vehicleType-{{ option.key }}"
                name="vehicleType"
                formControlName="vehicleType"
                type="radio"
                [value]="option.value"
              />
              <label class="govuk-label govuk-radios__label" for="vehicleType-{{ option.key }}">
                {{ option.key }}
              </label>
            </div>

            <div class="govuk-form-group">
              <label for="manufactureYear" class="govuk-label govuk-!-font-weight-bold">
                Year of manufacture
              </label>
              <input
                id="manufactureYear"
                type="number"
                name="manufactureYear"
                formControlName="manufactureYear"
                pattern="[0-9]*"
                class="govuk-input govuk-input--width-4"
              />
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</div>
