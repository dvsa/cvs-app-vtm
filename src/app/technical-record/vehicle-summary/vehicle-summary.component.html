<dl class="govuk-summary-list">
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Vehicle type
    </dt>
    <dd id="test-vehicleType" class="govuk-summary-list__value">
      {{ activeRecord.vehicleType | DefaultNullOrEmpty | uppercase }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Date of first registration
    </dt>
    <dd id="test-regnDate" class="govuk-summary-list__value">
      <span *ngIf="activeRecord.regnDate; else hyphen">
        {{ activeRecord.regnDate | date: 'dd/MM/yyyy' }}
      </span>
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">
      First use date
    </dt>
    <dd id="test-firstUseDate" class="govuk-summary-list__value">
      <span *ngIf="activeRecord.firstUseDate; else hyphen">
        {{ activeRecord.firstUseDate | date: 'dd/MM/yyyy' }}
      </span>
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Year of manufacture
    </dt>
    <dd id="test-manufactureYear" class="govuk-summary-list__value">
      {{ activeRecord.manufactureYear | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Number of axles
    </dt>
    <dd id="test-noOfaxles" class="govuk-summary-list__value">
      {{ activeRecord.noOfAxles | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      DTP number
    </dt>
    <dd id="test-dtpNumber" class="govuk-summary-list__value">
      <div *ngIf="activeRecord.brakes; else hyphen">
        {{ activeRecord.brakes.dtpNumber | DefaultNullOrEmpty }}
      </div>
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Axles fitted with a parking brake
    </dt>
    <dd class="govuk-summary-list__value" id="test-parkingBrakeMrk">
      <ng-container *ngIf="axlesHasParkingBrakeMrk(); else hyphen">
        <div *ngFor="let axel of activeRecord.axles">
          <div *ngIf="axel.parkingBrakeMrk">
            <span id="{{ 'test-parkingBrakeMrkAxle-' + axel.axleNumber }}"
              >Axle {{ axel.axleNumber }}</span
            >
          </div>
        </div>
      </ng-container>
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">
      Suspension type
    </dt>
    <dd id="test-suspensionType" class="govuk-summary-list__value">
      {{ activeRecord.suspensionType | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Road friendly suspension
    </dt>
    <dd id="test-roadFriendly" class="govuk-summary-list__value">
      {{ activeRecord.roadFriendly | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Vehicle class
    </dt>
    <dd id="test-vehicleClassDescription" class="govuk-summary-list__value">
      {{ vehicleClassDescription }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Vehicle configuration
    </dt>
    <dd id="test-vehicleConfiguration" class="govuk-summary-list__value">
      {{ activeRecord.vehicleConfiguration | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">
      Coupling type
    </dt>
    <dd id="test-couplingType" class="govuk-summary-list__value">
      {{ activeRecord.couplingType | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">
      Max load on coupling
    </dt>
    <dd id="test-maxLoadOnCoupling" class="govuk-summary-list__value">
      {{ activeRecord.maxLoadOnCoupling | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">
      Frame description
    </dt>
    <dd id="test-frameDescription" class="govuk-summary-list__value">
      {{ activeRecord.frameDescription | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Speed limiter exempt
    </dt>
    <dd id="test-speedLimiterMrk" class="govuk-summary-list__value">
      {{ activeRecord.speedLimiterMrk | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Tacho exempt
    </dt>
    <dd id="test-tachoExemptMrk" class="govuk-summary-list__value">
      {{ activeRecord.tachoExemptMrk | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Euro standard
    </dt>
    <dd id="test-euroStandard" class="govuk-summary-list__value">
      {{ activeRecord.euroStandard | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Fuel/propulsion system
    </dt>
    <dd id="test-fuelpropulsionsystem" class="govuk-summary-list__value">
      {{ activeRecord.fuelPropulsionSystem | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType == 'hgv'">
    <dt class="govuk-summary-list__key">
      Drawbar coupling fitted
    </dt>
    <dd id="test-drawbarCouplingFitted" class="govuk-summary-list__value">
      {{ activeRecord.drawbarCouplingFitted | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'hgv'">
    <dt class="govuk-summary-list__key">
      Offroad vehicle
    </dt>
    <dd id="test-offRoad" class="govuk-summary-list__value">
      {{ activeRecord.offRoad | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Number of wheels driven
    </dt>
    <dd id="test-numberOfWheelsDriven" class="govuk-summary-list__value">
      {{ activeRecord.numberOfWheelsDriven | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      EU vehicle category
    </dt>
    <dd id="test-euVehicleCategory" class="govuk-summary-list__value">
      {{ activeRecord.euVehicleCategory | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div
    class="govuk-summary-list__row"
    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
  >
    <dt class="govuk-summary-list__key">
      Emission limit
    </dt>
    <dd id="test-emissionsLimit" class="govuk-summary-list__value">
      {{ activeRecord.emissionsLimit | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Departmental vehicle marker
    </dt>
    <dd id="test-departmentalVehicleMarker" class="govuk-summary-list__value">
      {{ activeRecord.departmentalVehicleMarker | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Alteration marker
    </dt>
    <dd id="test-alterationMarker" class="govuk-summary-list__value">
      {{ activeRecord.alterationMarker | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
</dl>

<ng-container *ngIf="activeRecord.vehicleType === 'psv'">
  <vtm-vehicle-summary-psv [activeRecord]="activeRecord"></vtm-vehicle-summary-psv>
</ng-container>

<dl class="govuk-summary-list">
  <h2 class="govuk-heading-m">Type Approval</h2>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Approval type</dt>
    <dd id="test-approvalType" class="govuk-summary-list__value">
      {{ activeRecord.approvalType | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Approval type number</dt>
    <dd id="test-approvalTypeNumber" class="govuk-summary-list__value">
      {{ activeRecord.approvalTypeNumber | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">National type approval number</dt>
    <dd id="test-ntaNumber" class="govuk-summary-list__value">
      {{ activeRecord.ntaNumber | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Variant number</dt>
    <dd id="test-variantNumber" class="govuk-summary-list__value">
      {{ activeRecord.variantNumber | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Variant version number</dt>
    <dd id="test-variantVersionNumber" class="govuk-summary-list__value">
      {{ activeRecord.variantVersionNumber | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
    <dt class="govuk-summary-list__key">
      COIF serial number
    </dt>
    <dd id="test-vehicleSummary-coifSerialNumber" class="govuk-summary-list__value">
      {{ activeRecord.coifSerialNumber | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
    <dt class="govuk-summary-list__key">
      COIF certifier name
    </dt>
    <dd id="test-vehicleSummary-coifCertifierName" class="govuk-summary-list__value">
      {{ activeRecord.coifCertifierName | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>

  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
    <dt class="govuk-summary-list__key">
      COIF date
    </dt>
    <dd id="test-vehicleSummary-coifDate" class="govuk-summary-list__value">
      {{ activeRecord.coifDate | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
</dl>

<ng-template #hyphen>
  <span>-</span>
</ng-template>
