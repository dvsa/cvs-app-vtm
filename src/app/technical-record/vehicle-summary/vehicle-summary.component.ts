import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TechRecordHelpersService } from '@app/technical-record/tech-record-helpers.service';

@Component({
  selector: 'vtm-vehicle-summary',
  templateUrl: './vehicle-summary.component.html',
  styleUrls: ['./vehicle-summary.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleSummaryComponent implements OnInit {
  @Input() activeRecord: any;

  constructor(public techRecHelpers: TechRecordHelpersService) {}

  ngOnInit() {}

  axlesHasNoParkingBrakeMrk(axles): boolean {
    let baxlesHasNoParkingBrakeMrk = true;
    axles.forEach((axle) => {
      if (axle.parkingBrakeMrk === true) {
        baxlesHasNoParkingBrakeMrk = false;
      }
    });
    return baxlesHasNoParkingBrakeMrk;
  }
}
