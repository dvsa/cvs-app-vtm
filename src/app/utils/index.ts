export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


