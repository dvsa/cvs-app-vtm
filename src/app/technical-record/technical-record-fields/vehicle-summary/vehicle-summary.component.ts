import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
import { TechnicalRecordFieldsComponent } from '../technical-record-fields.component';
import { FormGroup, FormControl } from '@angular/forms';
import { TechRecord } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-vehicle-summary-fields',
  templateUrl: './vehicle-summary.component.html',
  styleUrls: ['./vehicle-summary.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleSummaryFieldsComponent extends TechnicalRecordFieldsComponent implements OnInit  {
  technicalRecord: FormGroup;
  options = { ['HGV']: 'hgv', ['PSV']: 'psv', ['Trailer']: 'trl' };

  ngOnInit() {
    this.technicalRecord = super.setUp();

    const summary: TechRecord = {} as TechRecord;

    this.technicalRecord.addControl('vehicleType', this.fb.control(''));
    this.technicalRecord.addControl('regnDate', this.fb.control(''));
    this.technicalRecord.addControl('manufactureYear', this.fb.control(''));
    this.technicalRecord.addControl('noOfAxles', this.fb.control(''));
    this.technicalRecord.addControl('dtpNumber', this.fb.control(''));


    this.technicalRecord.addControl('speedLimiterMrk', this.fb.control(''));
    this.technicalRecord.addControl('tachoExemptMrk', this.fb.control(''));
    this.technicalRecord.addControl('euroStandard', this.fb.control(''));
    this.technicalRecord.addControl('fuelPropulsionSystem', this.fb.control(''));
    this.technicalRecord.addControl('roadFriendly', this.fb.control(''));
    this.technicalRecord.addControl('drawbarCouplingFitted', this.fb.control(''));


    this.technicalRecord.addControl('vehicleConfiguration', this.fb.control(''));
    this.technicalRecord.addControl('offRoad', this.fb.control(''));
    this.technicalRecord.addControl('numberOfWheelsDriven', this.fb.control(''));
    this.technicalRecord.addControl('euVehicleCategory', this.fb.control(''));
    this.technicalRecord.addControl('emissionsLimit', this.fb.control(''));
    this.technicalRecord.addControl('departmentalVehicleMarker', this.fb.control(''));
    this.technicalRecord.addControl('alterationMarker', this.fb.control(''));

  }

}
