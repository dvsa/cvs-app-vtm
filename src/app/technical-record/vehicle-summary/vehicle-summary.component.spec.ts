import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA, Component, Input } from '@angular/core';

import { VehicleSummaryComponent } from '@app/technical-record/vehicle-summary/vehicle-summary.component';
import { TechRecord } from '@app/models/tech-record.model';
import { SharedModule } from '@app/shared/shared.module';
import { TESTING_UTILS } from '../../utils/testing.utils';

describe('VehicleSummaryComponent', () => {
  let component: VehicleSummaryComponent;
  let fixture: ComponentFixture<VehicleSummaryComponent>;
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [VehicleSummaryComponent, TestVehicleSummaryEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();

    fixture = TestBed.createComponent(VehicleSummaryComponent);
    component = fixture.componentInstance;
    component.activeRecord = TESTING_UTILS.mockTechRecord({
      approvalType: 'approval',
      variantNumber: '123',
      ntaNumber: '4566',
      vehicleClass: {
        code: '2',
        description: 'motorbikes over 200cc'
      }
    });
    component.activeRecord.axles = [TESTING_UTILS.mockAxle()];
  });

  it('should create view only with populated data', () => {
    component.editState = false;
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  it('should check if axles has no parking brake mrk', () => {
    expect(component.axlesHasParkingBrakeMrk()).toBeTruthy();
  });

  it('should create format the vehicle class description', () => {
    fixture.detectChanges();
    expect(component.vehicleClassDescription).toEqual('Motorbikes over 200cc');
  });

  it('should render the editable component if editState is true', () => {
    component.editState = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'vtm-vehicle-summary-edit',
  template: `
    <div>active record is: {{ techRecord | json }}</div>
  `
})
class TestVehicleSummaryEditComponent {
  @Input() techRecord: TechRecord;
}
