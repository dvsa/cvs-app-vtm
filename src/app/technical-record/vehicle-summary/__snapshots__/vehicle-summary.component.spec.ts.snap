// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleSummaryComponent should create view only with populated data 1`] = `
<vtm-vehicle-summary
  activeRecord={[Function Object]}
  editState="false"
  vehicleClassDescription={[Function String]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Vehicle type 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-vehicleType"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Date of first registration 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-regnDate"
      >
        
        <span>
          -
        </span>
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
     - 
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Year of manufacture 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-manufactureYear"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Number of axles 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-noOfaxles"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         DTP number 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-dtpNumber"
      >
        
        <span>
          -
        </span>
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Axles fitted with a parking brake 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-parkingBrakeMrk"
      >
        
        
        
        <div>
          
          <div>
            <span
              id="test-parkingBrakeMrkAxle-1"
            >
              Axle 1
            </span>
          </div>
        </div>
        
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Road friendly suspension 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-roadFriendly"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Vehicle class 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-vehicleClassDescription"
      >
         Motorbikes over 200cc 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Vehicle configuration 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-vehicleConfiguration"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    
    
    
    
    
    
    
    
    
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         EU vehicle category 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-euVehicleCategory"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Departmental vehicle marker 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-departmentalVehicleMarker"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Alteration marker 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-alterationMarker"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl><dl
    class="govuk-summary-list"
  >
    <h2
      class="govuk-heading-m"
    >
      Type Approval
    </h2>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Approval type
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-approvalType"
      >
         approval 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Approval type number
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-approvalTypeNumber"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        National type approval number
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-ntaNumber"
      >
         4566 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Variant number
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-variantNumber"
      >
         123 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Variant version number
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-variantVersionNumber"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    
    
    
  </dl>
</vtm-vehicle-summary>
`;

exports[`VehicleSummaryComponent should render the editable component if editState is true 1`] = `
<vtm-vehicle-summary
  activeRecord={[Function Object]}
  editState={[Function Boolean]}
  vehicleClassDescription={[Function String]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        <vtm-vehicle-summary-edit
          ng-reflect-tech-record="[object Object]"
        >
          <div>
            active record is: {
    "statusCode": "provisional",
    "adrDetails": {
      "vehicleDetails": {
        "approvalDate": "2020-03-10",
        "type": "centre axle battery"
      },
      "listStatementApplicable": false,
      "batteryListNumber": "",
      "declarationsSeen": false,
      "brakeDeclarationsSeen": false,
      "brakeDeclarationIssuer": "",
      "brakeEndurance": false,
      "weight": "",
      "compatibilityGroupJ": false,
      "documents": [],
      "permittedDangerousGoods": [
        "AT"
      ],
      "additionalExaminerNotes": "",
      "applicantDetails": {
        "name": "Ben",
        "street": "Robert green",
        "city": "Birmingham",
        "town": "lala land",
        "postcode": "NG4 12Z"
      },
      "memosApply": [],
      "additionalNotes": {
        "number": [],
        "guidanceNotes": []
      },
      "adrTypeApprovalNo": ""
    },
    "approvalType": "approval",
    "variantNumber": "123",
    "ntaNumber": "4566",
    "vehicleClass": {
      "code": "2",
      "description": "motorbikes over 200cc"
    },
    "axles": [
      {
        "parkingBrakeMrk": true,
        "axleNumber": 1,
        "brakes": {
          "brakeActuator": 123,
          "leverLength": 123,
          "springBrakeParking": true
        },
        "tyres": {
          "tyreSize": "16",
          "plyRating": "12S",
          "fitmentCode": "12313",
          "dataTrAxles": 1234
        },
        "weights": {
          "gbWeight": 100,
          "kerbWeight": 150,
          "designWeight": 250,
          "eecWeight": 150
        }
      }
    ]
  }
          </div>
        </vtm-vehicle-summary-edit>
      </dt>
    </div>
  </dl>
</vtm-vehicle-summary>
`;
